packages/core/ios/Sources/DdSdk.swift

```diff
@@ -16,38 +17,50 @@ func getDefaultAppVersion() -> String {
 
 @objc(DdSdk)
 class RNDdSdk: NSObject {
+    @objc var bridge: RCTBridge!
+
     @objc(requiresMainQueueSetup)
     static func requiresMainQueueSetup() -> Bool {
         return false
     }
     
+    let refreshRateListener: RefreshRateListener
     let mainDispatchQueue: DispatchQueueType
     @objc(methodQueue)
     let methodQueue: DispatchQueue = sharedQueue
     
+    private let jsLongTaskThresholdInSeconds: TimeInterval = 0.1;
+    
     convenience override init() {
-        self.init(mainDispatchQueue: DispatchQueue.main)
+        self.init(mainDispatchQueue: DispatchQueue.main, refreshRateListener: JSRefreshRateListener.init())
     }
     
-    init(mainDispatchQueue: DispatchQueueType) {
+    init(mainDispatchQueue: DispatchQueueType, refreshRateListener: RefreshRateListener) {
```

Comment:
You can initialize `RefreshRateListener` after knowing the configuration, so we don't monitor the vitals if it has been disabled 👍

Reply:
Yes, but in that case I am struggling to make it testable.
The best I have found is to pass a `RefreshRateListener.Type` in the init, but it makes the code much harder, and I still can't test it as I can't get the instance back when it's created.
I think the best way to do it is rather to not call `start` if vitals monitoring is disabled, what do you think?